gui/web/src/components/atoms/Button/Button.js

```diff
@@ -55,17 +52,27 @@ class Button extends Component {
       return
     }
     
+    if (this.props.eventName === "" || this.props.eventName === undefined) {
+      return
+    }
+    
     var _this = this
-    // TODO DS populate the eventData.
-    this._asyncRequests["sendMetricEvent"] = sendMetricEvent(this.props.baseUrl, this.props.eventName, {}).then(resp => {
+    var eventData = {
+      'eventName': this.props.eventName,
+      'type': 'generic',
+      'component': 'button'
+    }
+    this._asyncRequests["sendMetricEvent"] = sendMetricEvent(this.props.baseUrl, this.props.eventName, eventData).then(resp => {
       if (!_this._asyncRequests["sendMetricEvent"]) {
         // if it has been deleted it means we don't want to process the result
         return
       }
       
       delete _this._asyncRequests["sendMetricEvent"];
       
-      // TODO DS Determine how to process resp
+      if (resp["success"] === undefined) {
```

Comment:
use resp.success and resp.error instead of square bracket approach

specifically, here we should use `if (!resp.success)` so it checks for undefined and null. I think that should work, but will need to be tested. you can hardcode returning a failure on the backend to see if it logs. please add screenshot of the test. this can server as a substitute for an automated test for now.